packages/twenty-front/src/modules/object-metadata/schemas/createObjectMetadataItemPayloadSchema.ts

```diff
@@ -0,0 +1,31 @@
+import camelCase from 'lodash.camelcase';
+import { z, ZodTypeDef } from 'zod';
+
+import { objectMetadataItemSchema } from '@/object-metadata/schemas/objectMetadataItemSchema';
+import { CreateObjectInput } from '~/generated-metadata/graphql';
+
+type Input = Omit<
+  CreateObjectInput,
+  | 'imageIdentifierFieldMetadataId'
+  | 'labelIdentifierFieldMetadataId'
+  | 'namePlural'
+  | 'nameSingular'
+>;
+type Output = CreateObjectInput;
+
+export const createObjectMetadataItemPayloadSchema: z.ZodType<
+  Output,
+  ZodTypeDef,
+  Input
+> = objectMetadataItemSchema
+  .pick<Record<keyof Input, true>>({
+    description: true,
+    icon: true,
+    labelPlural: true,
+    labelSingular: true,
+  })
+  .transform((value) => ({
+    ...value,
+    nameSingular: camelCase(value.labelSingular),
```

Comment:
I feel this should not be here. The fact that we infer nameSingular from labelSingular is specific to the settings behavior and should not be part of object-metadata/ behaviors

Reply:
discussed it and this should be moved to settings

Reply:
Done, I moved this schema to settings and renamed it `settingsCreateObjectInputSchema`.